src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,7 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if (command.isMutating()) storage.save(addressBook);
```

Comment:
To improve readability, please follow the coding standards. (https://oss-generic.github.io/process/codingStandards/CodingStandard-Java.html#conditionals)